### PR TITLE
8356 builder features warning modal

### LINKED ISSUE #8356: Create warning modal when user comes back from the builder
Warning modal like we have with the "locked dataset/map", saying:

"You already have touched this map in the builder, if you edit it in the editor your map could break it."

### PR DESCRIPTION
**Pull request by javitonino**
Closes #8356

Checks for the presence of builder-only features(analyses, widgets and mapcaps) when opening a map, and displays this modal in case if it finds any:
screenshot_20160630_165322

### CODE REVIEW

**Comment by javitonino** [comment]
Please CR:
Backend: @gfiorav (sooooo many changes)
Frontend: @xavijam 
Text: @csobier

Thanks!

File: lib/assets/javascripts/cartodb/common/dialogs/builder_features_warning/template.jst.ejs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    The map may not display correctly and not all features will be available.<br />
+    It is recommended that you only edit this map using the Builder. It can be activated from your account settings.
+  </p>
+</div>
+<div class="Dialog-footer Dialog-footer--simple u-inner">
+  <button class="Button Button--secondary Dialog-footerBtn cancel">
+    <span>Back to dashboard</span>
+  </button>
+  <button class="js-ok Button Button--main">
+    <span>Continue (NOT RECOMMENDED)</span>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/builder_features_warning/template.jst.ejs]: Let's remove the not recommended :joy:

File: lib/assets/javascripts/cartodb/table/table.js [hunk 2]
```diff
@@ -168,6 +168,20 @@ $(function() {
         view.appendToBody();
       }
 
+      // *** Warning opening Builder maps in the old editor
+      if (this.vis.get('uses_builder_features')) {
+        var view = new cdb.editor.BuilderFeaturesWarningDialog({
+          //clean_on_hide: true,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/table.js]: uncomment this please.

**Comment by xavijam** [comment]
LGTM! :) :es:

File: lib/assets/javascripts/cartodb/table/table.js (hunk 2 above)
Comment (javitonino) [inline lib/assets/javascripts/cartodb/table/table.js]: Ooops, yeah :+1:

File: lib/assets/javascripts/cartodb/common/dialogs/builder_features_warning/template.jst.ejs (hunk 1 above)
Comment (javitonino) [inline lib/assets/javascripts/cartodb/common/dialogs/builder_features_warning/template.jst.ejs]: I was going for an aggresive line of writing, I may have gone too far, haha. Will wait for @csobier input and change the text, as surely there is more to change.

**Comment by csobier** [comment]
**WAIT! This map contains new Builder features that are not supported in the Editor**
If you continue, the map may not display correctly and some features will not be available.
It is highly recommended that you use the Builder to edit and build maps. 

**Tip**: Activate the Builder as your default tool from your Account settings.

**Comment by gfiorav** [comment]
Looks exactly like a 📦 to me :D

**Comment by javitonino** [comment]
Updated with some more awesomeness thanks to @csobier. Updated screenshot. Thanks everyone, will ship when :green_apple:

**Pull request merged**